core/object/object.cpp

```diff
@@ -795,14 +795,24 @@ Variant Object::call_const(const StringName &p_method, const Variant **p_args, i
 }
 
 void Object::notification(int p_notification, bool p_reversed) {
-	_notificationv(p_notification, p_reversed);
+	if (p_reversed) {
+		if (script_instance) {
+			script_instance->notification(p_notification, p_reversed);
+		}
+	} else {
+		_notificationv(p_notification, p_reversed);
+	}
 
-	if (script_instance) {
-		script_instance->notification(p_notification);
+	if (_extension && _extension->notification2) {
+		_extension->notification2(_extension_instance, p_notification, p_reversed);
```

Comment:
In order for old GDExtensions to continuing working, we'll need to still call `_extension->notification` if `_extension->notification2` is not set. So, something like:

```suggestion
	if (_extension) {
		if (_extension->notification2) {
			_extension->notification2(_extension_instance, p_notification, p_reversed);
		} else if (_extension->notification) {
			_extension->notification(_extension_instance, p_notification);
		}
```